stout/iterate.h

```diff
@@ -0,0 +1,136 @@
+#include <array>
+#include <iterator>
+
+#include "stout/context.h"
+#include "stout/loop.h"
+#include "stout/stream.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+template <typename Iter>
```

Comment:
s/Iter/Iterator/

Prefer full words please. Please update here and the rest of the review!